### DISCUSSION TITLE
Is it possible to merge button and stick combos together?

### DISCUSSION THREAD

**Discussion by Near2018** [ndeadly/MissionControl#435] [Q&A]
I wanted to make a button combo for my official pro controller which presses the left stick down, when x is pressed. 

Something like this:

```
   void SwitchController::ApplyButtonCombos(SwitchButtonData *buttons) {
        // Stick Combo = X
        if (buttons->X) {
            left_stick->down = 1;                                   
          
        }
```

How would a working line like this look like? 
Where would I have to place it?

Thanks in advance!

**Comment by ndeadly** [comment]
See https://github.com/ndeadly/MissionControl/discussions/427, this guy asked almost the exact same question

**Reply by Near2018** [reply]
Thanks! So the apply button combo method doesn't make much sense but how do I use buttons on the setdata method?

switch_report->input0x30.left_stick.SetData(STICK_ZERO, UINT12_MAX)
m_left_stick.SetData(STICK_ZERO, UINT12_MAX);

Where can I paste the pressed button?

**Reply by ndeadly** [reply]
> but how do I use buttons on the setdata method?

You don't. Those lines are the equivalent of writing something like `left_stick->down = 1; ` from your example for analog stick values. The stick values range from 0 to `UINT12_MAX` in X and Y. So fully stick fully pressed in one direction = 0 and fully pressed in the opposite direction = `UINT12_MAX`. `STICK_ZERO` is the neutral center position.

Since you need access to the stick values as well as the buttons, you either need to modify the `ApplyButtonCombos`  method to take a reference to the stick values too, or just add the code to `HandleIncomingReport`  [here](https://github.com/ndeadly/MissionControl/blob/0ee33b355b2d8649dddcb754ca928d15778636de/mc_mitm/source/controllers/emulated_switch_controller.cpp#L215) and also after [this line](https://github.com/ndeadly/MissionControl/blob/0ee33b355b2d8649dddcb754ca928d15778636de/mc_mitm/source/controllers/switch_controller.cpp#L90) if you also want it to work with official controllers. Eventually I will refactor these classes so it's only needed in one place.

Your code would look something like this

```
if (switch_report->input0x30.buttons.X ) {
    switch_report->input0x30.buttons.X = 0;
    switch_report->input0x30.left_stick.SetData(STICK_ZERO, UINT12_MAX)
}
```
